src/mrinufft/operators/autodiff.py

```diff
@@ -20,14 +20,15 @@ class _NUFFT_OP(torch.autograd.Function):
     @staticmethod
     def forward(ctx, x, traj, nufft_op):
         """Forward image -> k-space."""
-        ctx.save_for_backward(x, traj)
+        ctx.save_for_backward(x)
         ctx.nufft_op = nufft_op
+        ctx.dtype = traj.dtype
         return nufft_op.op(x)
 
     @staticmethod
     def backward(ctx, dy):
         """Backward image -> k-space."""
-        (x, traj) = ctx.saved_tensors
+        x = ctx.saved_tensors[0]
```

Comment:
Also, I fixed a minor bug in the code (which you would have n9oticed eventually)